core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -0,0 +1,230 @@
+            for (key, value) in touched_slots.into_iter().sorted_by_key(|(key, _)| *key) {
+                let hashed_key = key.hashed_key();
+                if previous_values[&hashed_key].unwrap_or_default() != value {
+                    let (initial_write_l1_batch_number, index) =
+                        l1_batches_for_initial_writes[&hashed_key];
+                    if initial_write_l1_batch_number == l1_batch_number {
+                        initial_writes.push(InitialStorageWrite {
+                            index,
+                            key: key.hashed_key_u256(),
+                            value,
+                        })
+                    } else {
+                        repeated_writes.push(RepeatedStorageWrite { index, value })
+                    }
+                }
+            }
+
+            CommitmentInput::PreBoojum {
+                common,
+                initial_writes,
+                repeated_writes,
+            }
+        } else {
+            let aux_commitments =
+                Self::calculate_aux_commitments(connection, header.number, protocol_version)
+                    .await?;
+
+            let mut state_diffs = Vec::new();
+            for (key, value) in touched_slots {
+                let hashed_key = key.hashed_key();
+                let prev_value = previous_values[&hashed_key].unwrap_or_default();
+                if prev_value != value {
+                    let (initial_write_l1_batch_number, index) =
+                        l1_batches_for_initial_writes[&hashed_key];
+                    if initial_write_l1_batch_number == l1_batch_number {
+                        state_diffs.push(StateDiffRecord {
+                            address: *key.address(),
+                            key: h256_to_u256(*key.key()),
+                            derived_key: StorageKey::raw_hashed_key(key.address(), key.key()),
+                            enumeration_index: 0u64,
+                            initial_value: U256::default(),
+                            final_value: h256_to_u256(value),
+                        });
+                    } else {
+                        state_diffs.push(StateDiffRecord {
+                            address: *key.address(),
+                            key: h256_to_u256(*key.key()),
+                            derived_key: StorageKey::raw_hashed_key(key.address(), key.key()),
+                            enumeration_index: index,
+                            initial_value: h256_to_u256(prev_value),
+                            final_value: h256_to_u256(value),
+                        });
+                    }
+                }
+            }
+            state_diffs.sort_unstable_by_key(|rec| (rec.address, rec.key));
+
+            CommitmentInput::PostBoojum {
+                common,
+                system_logs: header.system_logs,
+                state_diffs,
+                aux_commitments,
+            }
```

Comment:
would it make sense to extract this (and pre-bojum) to a method? Or maybe even a small method of `StateDiffRecord`?

Not sure it's worth it, just wanted you to consider this

Reply:
Makes sense

Reply:
Actually, just tried it and the code doesn't look much better, so I would probably leave it the current way